README.md

```diff
@@ -1,2 +1,91 @@
-# armada
+# Armada
 A python orchestrator for a installing, upgrading, and managing a collection of helm charts, dependencies, and values overrides.
+
+Note that this project is pre-alpha and under active development. It may undergo drastic changes to support the long-term vision but contributions are welcome. 
+
+# Overview
+
+TThe armada python library and command line tool provides a way to synchronize an operators intended state, consisting many several charts, dependencies, and overrides using a single file, with an intended helm target.  This allows operators to define many charts, different namespaces for those releases, and overrides in a central place and with a single command, deploy and upgrade them all where applicable.
+
+It supports fetching helm chart source from various local and remote locations, such as git/github endpoints.  In the future, it may supprot other mechanisms.
+
+It should also give the operator some indication of what is about to change by assisting with diffs for both values, values overrides, and template changes.
+
+Its functionality may extend beyond helm, assisting in interacting with kubernetes directly to perform basic pre and post steps, such as removing completed or failed jobs, or deleting resources that do not support upgrades.  However, primarily, it will be an interface to support orchestrating Helm.
```

Comment:
Given where Helm is today, I think Armada looks like a very solid approach for handling highly sophisticated applications or suites of applications. Excited to see how this develops!